src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strFromArr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const lastCheckSeparator = this[this.length - 1] === undefined
+      || this[this.length - 1] === null;
```

Comment:
why do you check pre last item? you already have the same check on line 14

Reply:
one of the tests crashed, when array has undifined at last element